test/integration/spec/localparticipant.js

```diff
@@ -758,60 +755,67 @@ describe('LocalParticipant', function() {
         thisRoom = await connect(thisToken, theseOptions);
-        thisParticipant = thisRoom.localParticipant;
-        await tracksPublished(thisParticipant, tracks.length);
+        const alice  = thisRoom.localParticipant;
+        await tracksPublished(alice, tracks.length);
 
         const thoseIdentities = identities.slice(1);
         const thoseTokens = thoseIdentities.map(getToken);
         const thoseOptions = Object.assign({ tracks: [] }, options);
 
-        thoseRooms = await waitFor(thoseTokens.map(thatToken => {
+        const [bobRoom, charlieRoom] = await waitFor(thoseTokens.map(thatToken => {
           return connect(thatToken, thoseOptions);
         }), `Rooms to get connected: ${sid}`);
+        thoseRooms = [bobRoom, charlieRoom];
 
-        await waitFor([thisRoom].concat(thoseRooms).map(room => {
+        await waitFor([thisRoom, bobRoom, charlieRoom].map(room => {
           return participantsConnected(room, identities.length - 1);
         }), `all Participants to get connected: ${sid}`);
 
-        thoseParticipants = thoseRooms.map(thatRoom => {
-          return thatRoom.participants.get(thisParticipant.sid);
+        const [aliceInBobRoom, aliceInCharlieRoom] = [bobRoom, charlieRoom].map(thatRoom => {
+          return thatRoom.participants.get(alice.sid);
         });
 
-        await waitFor(thoseParticipants.map(thatParticipant => {
-          return tracksSubscribed(thatParticipant, tracks.length);
-        }), `Track to get subscribed: ${sid}`);
+        const bobSubscribes =  tracksSubscribed(aliceInBobRoom, tracks.length);
+        const charlieSubscribes =  tracksSubscribed(aliceInCharlieRoom, tracks.length);
+
+        await waitFor(bobSubscribes, `Bob to subscribe Alice's track: ${sid}`);
+        await waitFor(charlieSubscribes, `Charlie to subscribe Alice's track: ${sid}`);
 
         if (when !== 'published') {
-          thisParticipant.unpublishTrack(thisTrack);
+          alice.unpublishTrack(thisTrack);
+
+          const bobSeesUnpublished =  tracksUnpublished(aliceInBobRoom, tracks.length - 1);
+          const charlieSeesUnpublished =  tracksUnpublished(aliceInCharlieRoom, tracks.length - 1);
 
-          await waitFor(thoseParticipants.map(thatParticipant => {
-            return tracksUnpublished(thatParticipant, tracks.length - 1);
-          }), `Track to get unpublished: ${sid}`);
+          await waitFor(bobSeesUnpublished, `Bob to see track getting Unpublished: ${sid}`);
+          await waitFor(charlieSeesUnpublished, `Charlie to see track getting Unpublished: ${sid}`);
 
           // NOTE(mmalavalli): Even though the "trackUnpublished" events are
           // fired on the RemoteParticipants, we need to make sure that the
           // SDP negotiation is complete before we re-publish the LocalTrack.
           // Therefore we wait for 2 seconds.
-          await waitForSometime(2000);
+          await waitForSometime(4000);
```

Comment:
As discussed over slack, Could you please create a bug for removing the wait? Also, please update the comment.

Reply:
Thanks @syerrapragada I have now filed: https://issues.corp.twilio.com/browse/JSDK-2976